File: app/src/main/java/com/example/igreeldownloader/util/bottomsheets/BroadcastService.java
package com.example.igreeldownloader.util.bottomsheets;

import static com.example.igreeldownloader.util.bottomsheets.Constants.downloadVideos;
import static com.example.igreeldownloader.util.bottomsheets.Utils.RootDirectoryFacebook;
import static com.example.igreeldownloader.util.bottomsheets.Utils.RootDirectoryInsta;
import static com.example.igreeldownloader.util.bottomsheets.Utils.RootDirectoryLikee;
import static com.example.igreeldownloader.util.bottomsheets.Utils.RootDirectoryMoz;
import static com.example.igreeldownloader.util.bottomsheets.Utils.RootDirectorySnapchat;

import android.app.DownloadManager;
import android.app.Service;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Environment;
import android.os.IBinder;
import android.util.Log;
import android.widget.Toast;

import androidx.annotation.Nullable;

import com.example.igreeldownloader.db.Database;
import com.example.igreeldownloader.models.FVideo;

public class BroadcastService extends Service {
    private static final String TAG = "BroadcastService";
    Database db;

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }

    @Override
    public void onCreate() {
        Log.d(TAG, "onCreate: broadcastService");
        db = Database.init(getApplicationContext());
        registerBroadcastRecerver();
    }

    private void registerBroadcastRecerver() {
        Log.d(TAG, "registerBroadcastRecerver: broadcast service");

        BroadcastReceiver downloadComplete = new BroadcastReceiver() {
            @Override
            public void onReceive(Context context, Intent intent) {
                long id = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, -1);

                if (downloadVideos.containsKey(id)) {
                    Log.d("receiver", "onReceive: download complete");

                    FVideo fVideo = db.getVideo(id);
                    String videoPath = null;

                    if (fVideo.getVideoSource() == FVideo.FACEBOOK) {

                        videoPath = Environment.getExternalStorageDirectory() +
                                "/Download" + RootDirectoryFacebook + fVideo.getFileName();

                    } else if (fVideo.getVideoSource() == FVideo.INSTAGRAM) {
                        videoPath = Environment.getExternalStorageDirectory() +
                                "/Download" + RootDirectoryInsta + fVideo.getFileName();
                    } else if (((FVideo) fVideo).getVideoSource() == FVideo.SNAPCHAT) {
                        videoPath = Environment.getExternalStorageDirectory() +
                                "/Download" + RootDirectorySnapchat + fVideo.getFileName();
                    } else if (fVideo.getVideoSource() == FVideo.LIKEE) {
                        videoPath = Environment.getExternalStorageDirectory() +
                                "/Download" + RootDirectoryLikee + fVideo.getFileName();
                    } else if (fVideo.getVideoSource() == FVideo.MOZ) {
                        videoPath = Environment.getExternalStorageDirectory() +
                                "/Download" + RootDirectoryMoz + fVideo.getFileName();
                    }

                    Toast.makeText(getApplicationContext(), "Download complete", Toast.LENGTH_SHORT).show();
                    db.updateState(id, FVideo.COMPLETE);
                    if (videoPath != null)
                        db.setUri(id, videoPath);
                    downloadVideos.remove(id);
                }
            }
        };

        registerReceiver(downloadComplete,
                new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE));
    }
}

File: app/src/main/java/com/bluelock/realdownloader/util/BroadcastService.kt
package com.bluelock.realdownloader.util

import android.app.DownloadManager
import android.app.Service
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.os.Environment
import android.os.IBinder
import android.util.Log
import android.widget.Toast
import com.bluelock.realdownloader.db.Database
import com.bluelock.realdownloader.models.FVideo

class BroadcastService : Service() {
    var db: Database? = null
    override fun onBind(intent: Intent): IBinder? {
        return null
    }

    override fun onCreate() {
        Log.d(TAG, "onCreate: broadcastService")
        db = Database.init(applicationContext)
        registerBroadcastRecerver()
    }

    private fun registerBroadcastRecerver() {
        Log.d(TAG, "registerBroadcastRecerver: broadcast service")
        val downloadComplete: BroadcastReceiver = object : BroadcastReceiver() {
            override fun onReceive(context: Context, intent: Intent) {
                val id = intent.getLongExtra(DownloadManager.EXTRA_DOWNLOAD_ID, -1)
                if (Constants.downloadVideos.containsKey(id)) {
                    Log.d("receiver", "onReceive: download complete")
                    val fVideo = db!!.getVideo(id)
                    var videoPath: String? = null
                    if (fVideo.videoSource == FVideo.FACEBOOK) {
                        videoPath = Environment.getExternalStorageDirectory().toString() +
                                "/Download" + Utils.RootDirectoryFacebook + fVideo.fileName
                    } else if (fVideo.videoSource == FVideo.INSTAGRAM) {
                        videoPath = Environment.getExternalStorageDirectory().toString() +
                                "/Download" + Utils.RootDirectoryInsta + fVideo.fileName
                    } else if ((fVideo as FVideo).videoSource == FVideo.SNAPCHAT) {
                        videoPath = Environment.getExternalStorageDirectory().toString() +
                                "/Download" + Utils.RootDirectorySnapchat + fVideo.fileName
                    } else if (fVideo.videoSource == FVideo.LIKEE) {
                        videoPath = Environment.getExternalStorageDirectory().toString() +
                                "/Download" + Utils.RootDirectoryLikee + fVideo.fileName
                    } else if (fVideo.videoSource == FVideo.MOZ) {
                        videoPath = Environment.getExternalStorageDirectory().toString() +
                                "/Download" + Utils.RootDirectoryMoz + fVideo.fileName
                    }
                    Toast.makeText(applicationContext, "Download complete", Toast.LENGTH_SHORT)
                        .show()
                    db!!.updateState(id, FVideo.COMPLETE)
                    if (videoPath != null) db!!.setUri(id, videoPath)
                    Constants.downloadVideos.remove(id)
                }
            }
        }
        registerReceiver(
            downloadComplete,
            IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE)
        )
    }

    companion object {
        private const val TAG = "BroadcastService"
    }
}